### PR TITLE
Reenabling uppercase BECH32 in QR codes

### PR DESCRIPTION
**Pull request by rockstardev**
Fix: #2110

### CODE REVIEW

**Review by Kukks — APPROVED** [review]
LGTM

**Comment by rockstardev** [comment]
Moving it to draft to signal it'll be re-enabled and merged once @NicolasDorier think we've raised enough awareness.

File: BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs [hunk 1]
```diff
@@ -76,15 +76,15 @@ class Prepare
...
+            if (bech32Prefixes.Any(a => model.BtcAddress.StartsWith(a, StringComparison.OrdinalIgnoreCase)))
+            {
+                model.InvoiceBitcoinUrlQR = model.InvoiceBitcoinUrlQR.Replace(
+                    $"bitcoin:{model.BtcAddress}", $"bitcoin:{model.BtcAddress.ToUpperInvariant()}",
+                    StringComparison.OrdinalIgnoreCase
+                );
+            }
         }
-        //private static string[] bech32Prefixes = new[] { "bc1", "tb1", "bcrt1" };
+        private static string[] bech32Prefixes = new[] { "bc1", "tb1", "bcrt1" };
```
Comment (NicolasDorier) [inline BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs:87]: Don't hardcode "bitcoin:" and the bech32 prefixes.

You have the scheme with this:
```
BTCPayNetworkProvider q;
((BTCPayNetwork)q.GetNetwork("BTC")).UriScheme
```
You have bech32 with... 

```
BTCPayNetworkProvider q;
                var hrp = Encoders.ASCII.EncodeData(((BTCPayNetwork)q.GetNetwork("BTC")).NBitcoinNetwork.GetBech32Encoder(NBitcoin.Bech32Type.WITNESS_PUBKEY_ADDRESS, false)
                    .HumanReadablePart);
```

Will make this code less ugly later on.

**Review by NicolasDorier — CHANGES_REQUESTED** [review]
Don't hardcode strings

File: BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs (hunk 1 above)
Comment (rockstardev) [reply BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs]: 616ceee

**Comment by rockstardev** [comment]
Tests are failing because of Bittrex not providing DASH_BTC rate any more. I see some other PRs are failing because of this as well. Let's address this as soon as possible @NicolasDorier @Kukks.

**Comment by Kixunil** [comment]
Awesome! Perhaps open a tracking issue for `BITCOIN:`?

**Comment by pavlenex** [comment]
Sure @Kixunil , however it's very unlikely we can push `BITCOIN:` any time soon, as there are lots of wallet that aren't yet compatible and as a payment software the last thing we want is customers not being able to pay an invoice.

**Comment by Kixunil** [comment]
Sure, I understand. (Although it's a bit worrying how many projects ignore standards).

**Comment by Kixunil** [comment]
Should the tracking issue be opened here or in discussions?

**Comment by gruve-p** [comment]
This broke our GRS implementation

File: BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs [hunk 2]
```diff
@@ -35,6 +37,12 @@ public class BitcoinLikePaymentHandler : PaymentMethodHandlerBase<DerivationSche
             _FeeRateProviderFactory = feeRateProviderFactory;
             _dashboard = dashboard;
             _WalletProvider = walletProvider;
+
+
+            var currentNetwork = (BTCPayNetwork)networkProvider.GetNetwork("BTC");
```
Comment (gruve-p) [inline BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs:42]: if we change `BTC` to `GRS` then our btcpayserver fork starts to work again

**Comment by pavlenex** [comment]
@Kixunil Let's open it as a discussion to keep a track of wallets we have to ping, once ready, we can move it as an issue (to-do) that can be implemented.

File: BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs (hunk 2 above)
Comment (rockstardev) [reply BTCPayServer/Payments/Bitcoin/BitcoinLikePaymentHandler.cs:42]: 30c7cbba96f55f08118fbba30888632bc56d8d1a thanks @Kukks

**Pull request merged**